src/datachain/lib/utils.py

```diff
@@ -28,3 +30,37 @@ def __init__(self, message):
 class DataChainColumnError(DataChainParamsError):
     def __init__(self, col_name, msg):
         super().__init__(f"Error for column {col_name}: {msg}")
+
+
+def normalize_col_names(col_names: Sequence[str]) -> dict[str, str]:
+    gen_col_counter = 0
+    new_col_names = {}
+    org_col_names = set(col_names)
+
+    for org_column in col_names:
+        new_column = org_column.lower()
+        new_column = re.sub("[^0-9a-z_\\s-]+", "_", new_column)
+        new_column = re.sub("[-_\\s]+", "_", new_column)
```

Comment:
Can't we simplify all that regular expression soup like this?

```suggestion
        new_column = re.sub("[^0-9a-z]+", "_", new_column)
```

Reply:
I believe it's the same: we're replacing one or more non-alphanumerical characters with a single underscore, and it's a greedy match.

```python
>>> re.sub("[^0-9a-z]+", "_", "test$-_-$string")
'test_string'
```